packages/cli/generators/discover/index.js

```diff
@@ -321,6 +328,46 @@ module.exports = class DiscoveryGenerator extends ArtifactGenerator {
       );
       debug(`Writing: ${fullPath}`);
 
+      if (this.options.relations) {
+        const relationImports = [];
+        const relationDestinationImports = [];
+        const foreignKeys = {};
+        for (const relationName in templateData.settings.relations) {
+          const relation = templateData.settings.relations[relationName];
+          const targetModel = this.artifactInfo.modelDefinitions.find(
+            model => model.name === relation.model,
+          );
+          // If targetModel is not in discovered models, skip creating relation
+          if (targetModel) {
+            Object.assign(templateData.properties[relation.foreignKey], {
+              relation,
+            });
+            relationImports.push(relation.type);
+            relationDestinationImports.push(relation.model);
+
+            foreignKeys[relationName] = {};
+            Object.assign(foreignKeys[relationName], {
+              name: relationName,
+              entity: relation.model,
+              entityKey: Object.entries(targetModel.properties).find(
+                x => x?.[1].id === 1,
+              )?.[0],
+              foreignKey: relation.foreignKey,
```

Comment:
Just a note for myself: We'll need to figure out how to handle composite foreign keys in the future.